user.go

```diff
@@ -39,6 +43,57 @@ func (a *App) GetUser(userID string) (*User, error) {
 	}
 }
 
+// GetUserByIdentifier gets a user using their identifier
+// returns user on success, error on failure
+func (a *App) GetUserByIdentifier(identifier string) (*User, error) {
+	var errorText string
+	message := "failed to get Passage User By Identifier"
+	limit := 1
+	res, err := a.client.ListPaginatedUsersWithResponse(
+		context.Background(),
+		a.ID,
+		&ListPaginatedUsersParams{
+			Limit:      &limit,
+			Identifier: &identifier,
+		},
+	)
+
+	if err != nil {
+		return nil, Error{Message: "network error:failed to get Passage User by Identifier" + message, err}
+	}
+
+	if res.JSON200 != nil {
+		users := res.JSON200.Users
+		if len(users) == 0 {
+			message = fmt.Sprintf(IdentifierDoesNotExist, identifier)
+			return nil, Error{
+				Message:    message,
+				StatusCode: http.StatusNotFound,
+				StatusText: http.StatusText(http.StatusNotFound),
+				ErrorText:  string(UserNotFound),
```

Comment:
This is a small distinction, but our error responses look like
```json
{
    "code": "user_not_found",
    "error": "User not found"
}
```
where `code` is a string from an enum that can be programatically switched on, while `error` is a human-readable string that can be displayed to the end user. The ErrorText field typically gets the `error` value from a response, but here it's being set to a code.

So even though it's wouldn't be set from a constant, I think it'd be more uniform with other errors to return
```suggestion
				ErrorText:  "User not found",
```

Reply:
Thanks for explaining the distinction. Made this change